core/src/worker/workflow/workflow_stream.rs

```diff
@@ -386,7 +386,7 @@ impl WFStream {
                 // We accept that there might be query tasks remaining in the buffer if we evicted
                 // and re-instantiated here. It's likely those tasks are now invalidated anyway.
                 if maybe_buffered.has_tasks() && should_evict {
-                    panic!("There were leftover buffered tasks");
+                    warn!("There were leftover buffered tasks when evicting run");
```

Comment:
Can you confirm -- you don't need to drain the buffer contents? So these are either going to be treated as valid items, or the buffer is about to be drained anyway, or destroyed?

Reply:
The tasks will just get dropped. It's not a meaningful issue for the reason the comment mentions. Worst case the task was valid, and it'll time out and get retried.